pyresample/test/test_sgeom/test_shapely.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2013-2022 Pyresample Developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Test cases for shapely."""
+import unittest
+
+import numpy as np
+from shapely.geometry import MultiPolygon, Polygon
+from shapely.ops import unary_union
+
+
+class TestShapely(unittest.TestCase):
+    """Test Shapely methods."""
+
+    def test_polygon_unary_union(self):
+        """Test MultiPolygon unary union."""
+        bounds1 = (-180.0, -90.0, 0.0, 90.0)  # extent1 = [-180, 0, -90, 90]
+        bounds2 = (0.0, 0.0, 180.0, 90.0)     # extent2 = [0, 180, -90, 90]
+        polygon1 = Polygon.from_bounds(*bounds1)
+        polygon2 = Polygon.from_bounds(*bounds2)
+        multipolygon = MultiPolygon([polygon1, polygon2])
+        unified_polygon = unary_union(multipolygon)
+        assert isinstance(unified_polygon, Polygon)
+        vertices = np.array(unified_polygon.exterior.coords)
+        expected_vertices = np.array([[-180., 90.],
+                                      [0., 90.],
+                                      [180., 90.],
+                                      [180., 0.],
+                                      [0., 0.],
+                                      [0., -90.],
+                                      [-180., -90.],
+                                      [-180., 90.]])
+        assert np.allclose(expected_vertices, vertices)
```

Comment:
This should give better error messages when things fail:

```suggestion
        np.testing.assert_allclose(expected_vertices, vertices)
```